fury/ui/helpers.py

```diff
@@ -23,6 +22,74 @@ def clip_overflow(textblock, width, side='right'):
     -------
     clipped text : str
         Clipped version of the text.
+    """
+    original_str = textblock.message
+    prev_bg = textblock.have_bg
+
+    is_overflowing, mid_ptr = check_overflow(textblock, width, '...', side)
+
+    if mid_ptr == 0:
+        return original_str
+
+    if not is_overflowing:
+        textblock.have_bg = prev_bg
+        return textblock.message
+
+
+def wrap_overflow(textblock, wrap_width, side='right'):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+    side : str, optional
+        Clips the overflowing text according to side.
+        It takes values "left" or "right".
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    prev_bg = textblock.have_bg
+
+    is_overflowing, mid_ptr = check_overflow(textblock, wrap_width, '', side)
+
+    if mid_ptr == 0:
+        return original_str
+
+    if not is_overflowing:
+        for i in range(len(original_str)):
+            if i % mid_ptr == 0 and i != 0:
+                original_str = original_str[:i]\
+                    + '\n' + original_str[i:]
+
+        textblock.have_bg = prev_bg
+        textblock.message = original_str
+        return textblock.message
+
```

Comment:
Same questions as above? what happens if we do not fill both conditions?

Reply:
Same as above, the checks were not necessary.